Pokemon Player/src/Battle.cpp

```diff
@@ -33,33 +42,38 @@ void Battle::fight()
 	{
 		//Read the current state of the battle
 		readRam();
+
+		//Update bost lead's stats
+		updatePoke(0);
+		updatePoke(-1);
 		//Calculate the best action
 		determineAction();
+
+		cout << nextMove[0] << " " << nextMove[1] << endl;
 		//Take the action
 		if(nextMove[0] == 1)
 		{
-			cout << "Attacking with move " << nextMove[1] << endl;
 			//Attack function
 			attack(nextMove[1]);
 		}
 
 		//Heal
-		
+		if(nextMove[0] == 3)
```

Comment:
3 is another magic number that could use an explanation:

http://en.wikipedia.org/wiki/Magic_number_%28programming%29#Unnamed_numerical_constants